master/pyext/src/validation/excludedvolume.py

```diff
@@ -1,125 +1,165 @@
+import math
+import os
+import csv
 
 
-class get_excluded_volume(get_input_information):
-    def __init__(self,mmcif_file):
+class GetExcludedVolume(GetInputInformation):
+    def __init__(self, mmcif_file):
         super().__init__(mmcif_file)
-        self.ID=str(get_input_information.get_id(self))
-        self.nos=get_input_information.get_number_of_models(self)
-        
-    def get_all_spheres (self,filetemp=None):
+        self.ID = str(GetInputInformation.get_id(self))
+        self.nos = GetInputInformation.get_number_of_models(self)
+        self.resultpath = '../static/results/'
+
+    def get_all_spheres(self, filetemp=None):
         """get information on all spheres for each model"""
         if filetemp is None:
-            Model_object=[b for i in self.system.state_groups for j in i for a in j for b in a]
-            model_dict={i+1:j._spheres for i,j in enumerate(Model_object)}
+            Model_object = [
+                b for i in self.system.state_groups for j in i for a in j for b in a]
+            model_dict = {i+1: j._spheres for i, j in enumerate(Model_object)}
         else:
             system, = ihm.reader.read(filetemp,
-                                  model_class=ihm.model.Model)
-            Model_object=[b for i in system.state_groups for j in i for a in j for b in a]
-            model_dict={i+1:j._spheres for i,j in enumerate(Model_object)}
-            #print (model_dict)
+                                      model_class=ihm.model.Model)
+            Model_object = [
+                b for i in system.state_groups for j in i for a in j for b in a]
+            model_dict = {i+1: j._spheres for i, j in enumerate(Model_object)}
+            # print (model_dict)
         return model_dict
-    
-    def get_nCr(self,n,r):
+
+    def get_nCr(self, n, r):
         """get all combinations"""
-        f=math.factorial
+        f = math.factorial
         return (f(n)/(f(r)*f(n-r)))
 
-    def get_violation_percentage(self,models_spheres_df:pd.DataFrame,viols:dict)->float:
+    def get_violation_percentage(self, models_spheres_df: pd.DataFrame, viols: dict) -> float:
         """get information on all spheres for each model"""
-        number_of_violations=sum(list(viols.values()))
-        number_of_combinations=self.get_nCr(models_spheres_df.shape[1],2)
+        number_of_violations = sum(list(viols.values()))
+        number_of_combinations = self.get_nCr(models_spheres_df.shape[1], 2)
         return (1-number_of_violations/number_of_combinations)*100
 
-    def get_violation_normalized(self,models_spheres_df:pd.DataFrame,viols:dict)->float:
+    def get_violation_normalized(self, models_spheres_df: pd.DataFrame, viols: dict) -> float:
         """ """
-        number_of_violations=sum(list(viols.values()))
-        normalization_constant=models_spheres_df.shape[1]*math.log(models_spheres_df.shape[1],10)
+        number_of_violations = sum(list(viols.values()))
```

Comment:
You don't need to coerce `values` to a `list`. `sum` will work on any iterable. (It's more efficient to not coerce to list since that makes another copy in memory.)